app/Models/Talk.php

```diff
@@ -148,6 +148,16 @@ public function getMp3Attribute()
         }
     }
 
+    public function setYoutubeIdAttribute($youtube_id)
+    {
+        if (strpos($youtube_id, 'youtu') !== false) {
+            preg_match('%(?:youtube(?:-nocookie)?\.com/(?:[^/]+/.+/|(?:v|e(?:mbed)?)/|.*[?&]v=)|youtu\.be/)([^"&?/ ]{11})%i', $youtube_id, $match);
```

Comment:
This regex filters the input content, and if it is a youtube URL it tries to parse and get only the ID. Otherwise, it keeps the original input.